app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -298,20 +305,47 @@ private void updateNavigationMenuSelection() {
         }
     }
 
-    private void addRoomColumns(HorizontalSnapScrollView scroller) {
-        LinearLayout root = (LinearLayout) scroller.getChildAt(0);
-        root.removeAllViews();
-        if (scroller.getColumnWidth() != 0) {
-            // update pre-calculated width with actual layout
-            columnWidth = scroller.getColumnWidth();
+    private void addRoomColumns(@NonNull HorizontalSnapScrollView horizontalScroller, @NonNull List<Lecture> lectures, int roomCount, boolean forceReload) {
+        int roomIndexLeft = horizontalScroller.getColumn();
+        int roomIndexRight = roomIndexLeft + maxRoomColumnsVisible;
```

Comment:
:abc: I have problems understanding what these variables mean. Can you describe it to me please? Maybe we come up with a better name then.

Reply:
roomIndexLeft = the index of the visible room most left on the screen.
roomIndexLeft = the index of the visible room most right on the screen.

Reply:
Example: The screen shows 4 rooms - roomIndexLeft = 0, roomIndexRight = 3. Then when I swipe to the right both indices are incremented by amount x. Do I understand correctly?

Reply:
If you swipe to right x columns, yes.

Reply:
Okay. Thank you. Maybe you can mention in a comment that the variables refer to the visible columns, like a sliding window.

Reply:
I think that `int roomIndexLeft = horizontalScroller.getColumn();` is explicit enough.